x/rewards/keeper/base_pay.go

```diff
@@ -54,7 +54,7 @@ func (k Keeper) SetAllBasePay(ctx sdk.Context, list []types.BasePayGenesis) {
 	}
 }
 
-func (k Keeper) popAllBasePayForChain(ctx sdk.Context, chainID string) (list []types.BasePayWithIndex) {
+func (k Keeper) popAllBasePayForChain(ctx sdk.Context, chainID string, removeAfterPop bool) (list []types.BasePayWithIndex) {
```

Comment:
let's separate pop and get, this implementation of them together is risky

Reply:
done